src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+
+    this.hidden = false;
+    this.class = 'herb';
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.class = 'carn';
+  }
+
+  bite(herb) {
+    if (!herb.hidden && herb.class !== 'carn') {
+      herb.health -= 50;
+
+      if (herb.health <= 0) {
+        Animal.alive = Animal.alive.filter((el) => el.health > 0);
```

Comment:
This filter operation is not efficient if called for each bite. Consider moving the removal of dead animals to a separate method and call it less frequently or when needed.